frontend2/src/components/Modal.tsx

```diff
@@ -0,0 +1,60 @@
+import { Transition, Dialog } from "@headlessui/react";
+import React, { Fragment } from "react";
+
+interface ModalProps {
+  isOpen: boolean;
+  closeModal: () => void;
+  title: string;
+  children?: React.ReactNode;
+}
+
+const Modal: React.FC<ModalProps> = ({
```

Comment:
I remember we discussed this but why did we decide to name the dialog modal again? Can't remember but OK with it regardless 👍

Reply:
My reasoning was so it wouldn't collide with the headless ui's naming of "Dialog" and because Popup might be confused with headless ui's popover

Reply:
Ok I remember now SGTM!